executor/point_get.go

```diff
@@ -186,7 +185,9 @@ func encodeIndexKey(e *baseExecutor, tblInfo *model.TableInfo, idxInfo *model.In
 	for i := range idxVals {
 		colInfo := tblInfo.Columns[idxInfo.Columns[i].Offset]
 		if colInfo.Tp == mysql.TypeString || colInfo.Tp == mysql.TypeVarString || colInfo.Tp == mysql.TypeVarchar {
-			idxVals[i], err = ranger.HandlePadCharToFullLength(sc, &colInfo.FieldType, idxVals[i])
+			var str string
+			str, err = idxVals[i].ToString()
```

Comment:
table.CastValue will append `0x0` if the string value's length is smaller than the BINARY column's length.
So here keep this if branch, don't use `table.CastValue` on string values

Reply:
Add this comment into code as well?